bio_hansel/subtyper.py

```diff
@@ -155,3 +170,370 @@ def subtype_reads(subtyping_params: SubtypingParams,
+                 file_path=input_fasta,
+                 scheme=scheme_name or scheme,
+                 scheme_version=scheme_version,
+                 scheme_subtype_counts=scheme_subtype_counts)
+
+    A = init_automaton(scheme_fasta)
+    df = find_in_fasta(A, input_fasta)
+
+    if df is None or df.shape[0] == 0:
+        logging.warning('No subtyping tile matches for input "%s" for scheme "%s"', input_fasta, scheme)
+        st.are_subtypes_consistent = False
+        return st, None
+
+    refpositions = [x for x, y in df.tilename.str.split('-')]
+    subtypes = [y for x, y in df.tilename.str.split('-')]
+    df['refposition'] = [int(x.replace('negative', '')) for x in refpositions]
+    df['subtype'] = subtypes
+    df['is_pos_tile'] = ~df.tilename.str.contains('negative')
+    logging.debug('df: %s', df)
+    dfpos = df[df.is_pos_tile]
+    logging.debug('dfpos: %s', dfpos)
+    subtype_lens = dfpos.subtype.apply(len)
+    max_subtype_strlen = subtype_lens.max()
+    logging.debug('max substype str len: %s', max_subtype_strlen)
+    dfpos_highest_res = dfpos[subtype_lens == max_subtype_strlen]
+    logging.debug('dfpos_highest_res: %s', dfpos_highest_res)
+    pos_subtypes = [[int(y) for y in x.split('.')] for x in dfpos.subtype.unique()]
+    pos_subtypes.sort(key=lambda a: len(a))
+    logging.debug('pos_subtypes: %s', pos_subtypes)
+    inconsistent_subtypes = find_inconsistent_subtypes(pos_subtypes)
+    logging.debug('inconsistent_subtypes: %s', inconsistent_subtypes)
+    st.n_tiles_matching_all = df.tilename.unique().size
+    st.n_tiles_matching_positive = dfpos.tilename.unique().size
+    st.n_tiles_matching_subtype = dfpos_highest_res.tilename.unique().size
+    pos_subtypes_str = [x for x in dfpos.subtype.unique()]
+    pos_subtypes_str.sort(key=lambda x: len(x))
+    st.all_subtypes = '; '.join(pos_subtypes_str)
+    subtype_list = [x for x in dfpos_highest_res.subtype.unique()]
+    st.subtype = '; '.join(subtype_list)
+    st.n_tiles_matching_all_expected = ';'.join([str(scheme_subtype_counts[x].all_tile_count) for x in subtype_list])
+    st.n_tiles_matching_positive_expected = ';'.join(
+        [str(scheme_subtype_counts[x].positive_tile_count) for x in subtype_list])
+    st.n_tiles_matching_subtype_expected = ';'.join(
+        [str(scheme_subtype_counts[x].subtype_tile_count) for x in subtype_list])
+    st.tiles_matching_subtype = '; '.join([x for x in dfpos_highest_res.tilename.unique()])
+
+    if len(inconsistent_subtypes) > 0:
+        st.are_subtypes_consistent = False
+        st.inconsistent_subtypes = inconsistent_subtypes
+
+    possible_downstream_subtypes = [s for s in scheme_subtype_counts
+                                    if re.search("^({})(\.)(\d)$".format(re.escape(st.subtype)), s)]
+    non_present_subtypes = []
+    if possible_downstream_subtypes:
+        for subtype in possible_downstream_subtypes:
+            blah = df['subtype']
+            # wtf is going on
```

Comment:
Remove debugging code

Reply:
Fixed in 271b892d1a7eac71c0089d8c57c0a946a87cd91b